scene/resources/texture.cpp

```diff
@@ -2413,7 +2413,7 @@ float GradientTexture2D::_get_gradient_offset_at(int x, int y) const {
 }
 
 void GradientTexture2D::set_width(int p_width) {
-	width = p_width;
+	width = MIN(p_width, 16384);
```

Comment:
IMO an `ERR_FAIL_INDEX` would be better, I don't like silently changing user input without letting them know. Same should be done for 1D.